node_script/wrapper.py

```diff
@@ -83,68 +129,15 @@ def inference_step(self, msg: Image) -> Tuple[Image, List[int], List[float], Opt
-        seg_img.header = self.header
 
         # Get class and score arrays
         class_indexes = instances.pred_classes.tolist()
         scores = instances.scores.tolist()
-        return seg_img, class_indexes, scores, visualized_output
-
-    def get_debug_img(self, visualized_output: VisImage) -> Image:
-        # Call after inference_step
-        debug_img = self.bridge.cv2_to_imgmsg(visualized_output.get_image(),
-                                              encoding="rgb8")
-        assert self.header is not None
-        debug_img.header = self.header
-        return debug_img
-
-    def get_debug_segimg(self) -> Image:
-        # Call after inference_step
-        assert self.data is not None
-        human_friendly_scaling = 255 // self.data.max()
-        new_data = (self.data * human_friendly_scaling).astype(np.uint8)
-        debug_seg_img = self.bridge.cv2_to_imgmsg(new_data, encoding="mono8")
-        assert self.header is not None
-        debug_seg_img.header = self.header
-        return debug_seg_img
-
-    def get_segmentation_info(self, seg_img: Image,
-                              detected_classes: List[int],
-                              scores: List[float]) -> SegmentationInfo:
-        detected_classes_names = [self.class_names[i] for i in detected_classes]
-        seg_info = SegmentationInfo(detected_classes=detected_classes_names,
-                                    scores=scores,
-                                    segmentation=seg_img,
-                                    header=self.header)
-        return seg_info
-
-    def get_label_array(self, detected_classes: List[int]) -> LabelArray:
-        # Label 0 is reserved for the background
-        labels = [Label(id=i + 1, name=self.class_names[i]) for i in detected_classes]
-        lab_arr = LabelArray(header=self.header,
-                             labels=labels)
-        return lab_arr
-
-    def get_score_array(self, scores: List[float]) -> VectorArray:
-        vec_arr = VectorArray(header=self.header,
-                              vector_dim=len(scores),
-                              data=scores)
-        return vec_arr
-
-    def infer(self, msg: Image) -> Tuple[SegmentationInfo, Optional[Image], Optional[Image]]:
-        seg_img, labels, scores, vis_img = self.inference_step(msg)
-        seg_info = self.get_segmentation_info(seg_img, labels, scores)
-
-        if self.node_config.out_debug_img:
-            debug_img = self.get_debug_img(vis_img)
-        else:
-            debug_img = None
-
-        if self.node_config.out_debug_img:
-            debug_seg_img = self.get_debug_segimg()
-        else:
-            debug_seg_img = None
-
-        return seg_info, debug_img, debug_seg_img
+        result = InferenceRawResult(
+            data,
+            class_indexes,
+            scores,
+            visualized_output,
+            msg.header,
+            self.class_names)
```

Comment:
Should we pass directly `[self.class_names[i] for i in class_indexes]`? 
Or do we really want to keep the full vocabulary?

Reply:
That's true. I will fix this